src/pure/Queue.mo

```diff
@@ -88,19 +104,56 @@ module {
   /// Runtime: `O(1)`.
   ///
   /// Space: `O(1)`.
-  public func size<T>(queue : Queue<T>) : Nat {
-    debug assert queue.1 == List.size(queue.0) + List.size(queue.2);
-    queue.1
+  public func size<T>(queue : Queue<T>) : Nat = switch queue {
+    case (#empty) 0;
+    case (#one(_)) 1;
+    case (#two(_, _)) 2;
+    case (#three(_, _, _)) 3;
+    case (#idles((l, nL), (r, nR))) {
+      debug assert Stacks.size(l) == nL and Stacks.size(r) == nR;
```

Comment:
I have a horrible feeling that dfx always enables --debug so these assertion will always run in user code.
Might be worth checking (build a default dfx project check if debug expressions have an effect)

Reply:
Or maybe Luc remembers what the status with dfx is.

Reply:
Oh no, [yes they are executed by default](https://internetcomputer.org/docs/motoko/main/reference/language-manual/#debug). People would need to pass `--release` flag. One would argue that every production program should be compiled with this flag. IMO it is good to have those invariants checked. Maybe I'll check which ones are actually expensive to compute...

Reply:
I'd prefer to leave the `debug assert` invariants for safety and a form of extra documentation.
`mops bench` uses now the Release flag by default. And users should be advised in the docs to use that flag in production.